src/main/java/upb/dice/twitter/RateLimitChecker.java

```diff
@@ -0,0 +1,29 @@
+package upb.dice.twitter;
+
+import twitter4j.RateLimitStatus;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+import java.util.Map;
+/**
+ * Checks the rate limit for the requests to determine the remaining limit for the search
+ */
+public class RateLimitChecker {
+    private Twitter twitter = TwitterFactory.getSingleton();
+    private int limitReached = 0;
+    /**
+     * This method checks if the rate limit has been reached for the current search
+     * @return 1 if the rate limit has reached
+     */
+    public int rateLimitCheck() throws TwitterException{
```

Comment:
I am not sure whether I understand the semantics of this method. It is clear that it should check the rate limit. But why does it return a `1` when the rate limit has been hit? And why does it return an `int` if it obviously returns a flag (i.e., a boolean value)?